### DISCUSSION TITLE
How to return single value in a query with one-to-one relationship and not an array

### DISCUSSION THREAD

**Discussion by evrazi** [supabase/supabase#9026] [Questions]
Found this to be a problem when querying data with multiple foreign keys that have deeply nested returns.

Simplified example:

Now
 `
 {
     "id": 1,
     "example_table1": [
         {
             example_table2: [
                 {
                     "key": "alyktm3cf11bypgg"
                 }
             ]
         }
     ]
 }
 `

What I am looking for
`
{
    "id": 1,
    "key": "alyktm3cf11bypgg"
}
`
or
`
{
    "id": 1,
    "example_table1": "alyktm3cf11bypgg"
}
`

So is it possible to tell supabase that the return should be a single value or an object?

Something like this. When `[0]` is added after the table name
`
    const { data, error } = await supabase
        .from('example_table')
        .select('*, 
            example_table1[0](
                example_table2[0](
                    key
                )
            )
        ') ;
`

**Comment by steve-chavez** [comment]
One-to-one relationships now avoid the sub array https://postgrest.org/en/stable/api.html#one-to-one-relationships. This version is not yet deployed on Supabase though.

Flattening and removing the object is not yet supported([ref](https://github.com/PostgREST/postgrest/issues/1233#issuecomment-1115118458)). For now that has to be done client side.

**Reply by evrazi** [reply]
Thanks.

Would you have an estimate when the feature will be available/if it will be available @steve-chavez ?
